### ISSUE TITLE
Please add localization support to the IDE

### ISSUE THREAD

**Issue by ajmcoqui** [rstudio/rstudio#4388]
### System details

    RStudio Edition : Desktop and Server
    RStudio Version : all
    OS Version      : all
    R Version       : all

### Describe the problem in detail
It is not currently possible to change the locale or language of the IDE such that the labels and other text are in something other than English.

### Describe the behavior you expected
I would like to be able to choose the language of the text of the IDE. The language that comes up most often is French, to support Canadian organizations with language accessibility requirements.

I know that RStudio is not responsible for localizing the R language itself, but being able to choose the language of the IDE labels, options, and descriptions would be useful.

**Comment by jmcphers** [comment]
See here for translations of the messages R emits: https://developer.r-project.org/Translations30.html

**Comment by samuel-rosa** [comment]
Yes, it would be very helpful to have the menus in languages other than English. I spend a lot of time explaining the meaning of words for my Brazilin students.

**Comment by phgrosjean** [comment]
Yes, it would be nice to have RStudio in a different language than English too.

**Comment by jmaspons** [comment]
I found several files like https://github.com/rstudio/rstudio/blob/main/src/node/desktop/src/assets/locales/fr.json and [others](https://github.com/search?q=repo%3Arstudio/rstudio%20path%3A_fr.&type=code).

Is it enough to add new files with new locale codes to add new languages?

**Comment by gtritchie** [comment]
> Yes, it would be nice to have RStudio in a different language than English too.

Most of the UI has been localized into French in addition to English. You can switch languages in Tools / Global Options / General / Advanced / User Interface Language.

<img width="632" alt="screenshot of global options showing location of language switching UI" src="https://github.com/rstudio/rstudio/assets/10569626/5b0c76bb-12c6-4b7d-9849-8b65f783902e">

**Comment by gtritchie** [comment]
> Is it enough to add new files with new locale codes to add new languages?

That is the first step. You would also need to tweak the user interface to present the new language option (it isn't automatic, unfortunately). For example, see:

https://github.com/rstudio/rstudio/blob/f74629cb577dd5aae12bb9b6fd0977a089189842/src/gwt/src/org/rstudio/studio/client/workbench/prefs/views/GeneralPreferencesPane.java#L370-L377

There are probably some other small tweaks needed, but I'd need to hunt those down.
